test/scan/test_scan_spmd.py

```diff
@@ -39,10 +39,21 @@ def fn(carry, x):
     # Check the input and output sharding. Note that we do this after
     # `torch_xla.sync()` to ensure the output tensors are materialized and
     # have taken on sharding annotations propagated by the compiler.
-    for tensor in [init, xs, final_carry, ys]:
-      self.assertIn('ShardingSpec: {devices=[',
+    N = xr.global_runtime_device_count()
+    for tensor in [init, final_carry]:
+      self.assertIn('ShardingSpec: {'
```

Comment:
nits: using `_get_xla_sharding_spec` and `_get_xla_op_sharding` instead of `_get_xla_tensor_debug_info`, easier to parse.

Reply:
TIL

Reply:
Turns out I can't do that because there's no Python bindings (e.g. comparator) exposed for xla::OpSharding at all.

Reply:
At least replaced sharding spec parsing with _get_xla_sharding_spec

Reply:
Oh right, `_get_xla_sharding_spec` returns a string, and `_get_xla_op_sharding` the actual xla::OpSharding binding. The former is confusing...